src/state.rs

```diff
@@ -136,6 +136,33 @@ pub fn advertised_node_for_surface(w: &WlSurface, dh: &DisplayHandle) -> Option<
     None
 }
 
+#[derive(Debug, Clone, Copy)]
+pub struct GestureState {
+    pub fingers: u32,
+    pub direction: Direction,
+    pub delta: f64,
+}
+
+impl GestureState {
+    pub fn new(fingers: u32) -> Self {
+        GestureState {
+            fingers,
+            direction: Direction::Right,
+            delta: 0.0,
+        }
+    }
+}
+
+impl Default for GestureState {
+    fn default() -> Self {
+        GestureState {
+            fingers: 0,
+            direction: Direction::Right,
+            delta: 0.0,
+        }
```

Comment:
```suggestion
        GestureState::new(0)
```

Reply:
Fixed in 7385acea4cf6bc50834d253a190a30d532feaa64